### PR TITLE
improvement(lint): 'TranslationTypo' - link to Crowdin 

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description

A discussion on https://github.com/ankidroid/Anki-Android/pull/17160 prompted the fact that we can save time by linking directly to Crowdin if errors occur

Sample link: https://crowdin.com/editor/ankidroid/7290/en-af#q=create_subdeck

## Approach
* obtain a 'file' on Crowdin:
  * map from `01-core.xml` to `7290`
  * map languages: `zh-rCN` to `zhcn`
* query the string in the file: `#search_string` in the URL fragment works

## How Has This Been Tested?
Manual/unit tests

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by lukstbit — APPROVED** [review]
This is fantastic!
Left some nitpicks don't consider them blocking.

**Review by mikehardy — APPROVED** [review]
This should speed up the "crowdin fix cycle" a lot, very cool

Hopefully all the embedded mapping logic (upstream file numbers, URL structure, language code mapping and stuff) is stable so this doesn't need much upkeep. If that bears out in the future then this is magic

**Comment by mikehardy** [comment]
lukstbit approved but with some non-blocking comments - I don't have anything to add to those so marked as "good to merge" but waiting for feedback

File: lint-rules/src/test/java/com/ichi2/anki/lint/utils/CrowdinTest.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            "values-zu" to "zu"
+        )
+
+        val fileIdentifier = CrowdinFileIdentifier(7290)
+
+        val stringsToTest = mutableListOf<String>()
+        for ((folderName, expected) in folderNames) {
+            val languageTag = CrowdinLanguageTag.fromFolderName(folderName)!!
+
+            if (expected != null) {
```
[the remark this answers is not in the document]
Comment (david-allison) [reply lint-rules/src/test/java/com/ichi2/anki/lint/utils/CrowdinTest.kt:133]: I've updated the documentation.

This allows a user to provide `"values-af" to null`, and the routine prints out a value for a user to manually check. In this case: https://crowdin.com/editor/ankidroid/7290/en-af#q=send_feedback

[the remark this answers is not in the document]
Comment (david-allison) [reply lint-rules/src/test/java/com/ichi2/anki/lint/utils/CrowdinTest.kt]: @lukstbit comment is waiting on your acceptance. 

```kotlin

    @Test
    fun manuallyTestLinks() {
        // when adding a new string, define it as 'null' and this test will
        // provide a URL for you to manually test:
        // example: "values-xx" to null
        // Once the URL works, replace 'null' with the language Crowdin uses
```

**Pull request merged**